com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeReferenceVolumeAuthoringEditor.cs

```diff
@@ -165,12 +198,141 @@ private void Constrain()
+                {
+                    List<Matrix4x4> probeBuffer = new List<Matrix4x4>();
+                    List<int> probeMap = new List<int>();
 
-            if (Event.current.type == EventType.Layout)
-                pvra.DrawProbeGizmos();
+                    for (int i = 0; i < cell.probePositions.Length; i++)
+                    {
+                        // Skip probes which aren't of current brick size
+                        if (cell.bricks[i / 64].size == brickSize)
+                        {
+                            probeBuffer.Add(Matrix4x4.TRS(cell.probePositions[i], Quaternion.identity, Vector3.one * (0.3f * (brickSize + 1))));
+                            probeMap.Add(i);
+                        }
+
+                        // Batch limit reached or out of probes
+                        if (probeBuffer.Count >= probesPerBatch || i == cell.probePositions.Length - 1)
+                        {
+                            MaterialPropertyBlock prop = new MaterialPropertyBlock();
+                            float gradient = largestBrickSize == 0 ? 1 : brickSize / largestBrickSize;
+                            prop.SetColor("_Color", Color.Lerp(Color.red, Color.green, gradient));
+                            props.Add(prop);
+
+                            probeBuffers.Add(probeBuffer.ToArray());
+                            probeBuffer = new List<Matrix4x4>();
+                            probeMaps.Add(probeMap.ToArray());
+                            probeMap = new List<int>();
+                        }
+                    }
+                }
+
+                var debugData = new CellInstancedDebugProbes();
+                debugData.probeBuffers = probeBuffers;
+                debugData.props = props;
+                debugData.probeMaps = probeMaps;
+                debugData.cellPosition = cell.position;
+
+                Vector4[][] shBuffer = new Vector4[4][];
+                for (int i = 0; i < shBuffer.Length; i++)
+                    shBuffer[i] = new Vector4[probesPerBatch];
+
+                Vector4[] validityColors = new Vector4[probesPerBatch];
+
+                for (int batchIndex = 0; batchIndex < debugData.probeMaps.Count; batchIndex++)
+                {
+                    for (int indexInBatch = 0; indexInBatch < debugData.probeMaps[batchIndex].Length; indexInBatch++)
+                    {
+                        int probeIdx = debugData.probeMaps[batchIndex][indexInBatch];
+
+                        shBuffer[0][indexInBatch] = new Vector4(cell.sh[probeIdx][0, 3], cell.sh[probeIdx][0, 1], cell.sh[probeIdx][0, 2], cell.sh[probeIdx][0, 0]);
+                        shBuffer[1][indexInBatch] = new Vector4(cell.sh[probeIdx][1, 3], cell.sh[probeIdx][1, 1], cell.sh[probeIdx][1, 2], cell.sh[probeIdx][1, 0]);
+                        shBuffer[2][indexInBatch] = new Vector4(cell.sh[probeIdx][2, 3], cell.sh[probeIdx][2, 1], cell.sh[probeIdx][2, 2], cell.sh[probeIdx][2, 0]);
```

Comment:
I see we still do L1 all the time, wonder if it is worth adding L2 when we are in that mode

Reply:
Considering that the L2 mode is completely broken right now, I won't be adding it in this PR. but that's an improvement to consider in the future. Also, it would be necessary to link it with the setting we have in HDRP, so the gizmo is in sync with either L1 or L2 used in the rendering.